Sources/FoundationEssentials/FileManager/FileManager+Directories.swift

```diff
@@ -489,22 +489,10 @@ extension _FileManagerImpl {
     
     var currentDirectoryPath: String? {
 #if os(Windows)
-        var dwLength: DWORD = GetCurrentDirectoryW(0, nil)
-        guard dwLength > 0 else { return nil }
-
-        for _ in 0 ... 8 {
-            if let szCurrentDirectory = withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(dwLength), {
-                let dwResult: DWORD = GetCurrentDirectoryW(dwLength, $0.baseAddress)
-                if dwResult == dwLength - 1 {
-                    return String(decodingCString: $0.baseAddress!, as: UTF16.self)
-                }
-                dwLength = dwResult
-                return nil
-            }) {
-                return szCurrentDirectory
-            }
+        let initialSize = GetCurrentDirectoryW(0, nil)
```

Comment:
I think that we should guard that the returned value is `>= 0`. Also, most of the windows code that works with the system apis follows the windows recommendations (e.g. would name this `dwSize`).

Reply:
Done.